packages/rollup-plugin-workbox/tsconfig.json

```diff
@@ -0,0 +1,22 @@
+// Don't edit this file directly. It is generated by /scripts/update-package-configs.ts
+
+{
+  "extends": "../../tsconfig.node-base.json",
+  "compilerOptions": {
+    "module": "commonjs",
+    "outDir": "./dist",
+    "rootDir": "./src",
+    "composite": true,
+    "allowJs": true,
+    "lib": ["webworker"],
+    "types": [
+      "workbox-build"
```

Comment:
this will be overwritten when we re-generate these scripts. this is needed only when this defined global types without importing them. but it looks like you're importing them

Reply:
what is the error you're getting if you don't have this?

Reply:
Ah alright, I see

I also had to add `"lib": ["webworker"]` for it to work, because otherwise the `FetchEvent` type wouldnt be recognized by TS. How should we go about that?

Reply:
Yeah that's a good one, that's for the service worker file? Need to check if we can do that per-file.

Reply:
We could try a reference: https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-lib-

Otherwise we need to look into a way of customizing the tsconfig for individual packages

Reply:
I couldnt get the references thing to work, so I changed the `update-package-configs.js` script to allow for a `tsconfig.override.json` here: https://github.com/modernweb-dev/web/pull/378/files#diff-38c13ff8b2dcbf2b25b2d026673fb386R65

But... now I'm getting a whole bunch of other TS errors, with seemingly the exact same config I had before 🙃 

```
../../../../.nvm/versions/node/v13.14.0/lib/node_modules/typescript/lib/lib.webworker.d.ts:25:1 - error TS6200: Definitions of the following identifiers conflict with those in another file: EventListenerOrEventListenerObject, ImportExportKind, TableKind, ValueType, ExportValue, Exports, ImportValue, ModuleImports, Imports, HeadersInit, BodyInit, RequestInfo, BlobPart, DOMHighResTimeStamp, CanvasImageSource, OffscreenRenderingContext, MessageEventSource, ImageBitmapSource, TimerHandler, PerformanceEntryList, ReadableStreamReadResult, VibratePattern, AlgorithmIdentifier, HashAlgorithmIdentifier, BigInteger, NamedCurve, GLenum, GLboolean, GLbitfield, GLint, GLsizei, GLintptr, GLsizeiptr, GLuint, GLfloat, GLclampf, TexImageSource, Float32List, Int32List, GLint64, GLuint64, Uint32List, BufferSource, DOMTimeStamp, FormDataEntryValue, IDBValidKey, Transferable, BinaryType, CanvasDirection, CanvasFillRule, CanvasLineCap, CanvasLineJoin, CanvasTextAlign, CanvasTextBaseline, ClientTypes, EndingType, IDBCursorDirection, IDBRequestReadyState, IDBTransactionMode, ImageSmoothingQuality, KeyFormat, KeyType, KeyUsage, NotificationDirection, NotificationPermission, OffscreenRenderingContextId, PermissionName, PermissionState, PushEncryptionKeyName, PushPermissionState, ReferrerPolicy, RequestCache, RequestCredentials, RequestDestination, RequestMode, RequestRedirect, ResponseType, ServiceWorkerState, ServiceWorkerUpdateViaCache, VisibilityState, WebGLPowerPreference, WorkerType, XMLHttpRequestResponseType

25 interface AddEventListenerOptions extends EventListenerOptions {
   ~~~~~~~~~

    Conflicts are in this file.

../../../../.nvm/versions/node/v13.14.0/lib/node_modules/typescript/lib/lib.webworker.d.ts:89:5 - error TS2687: All declarations of 'privateKey' must have identical modifiers.

89     privateKey?: CryptoKey;
       ~~~~~~~~~~

etc etc etc
```

Reply:
Hrm okay, locally I'm now randomly getting either 19 tsc errors (like above here), and sometimes only these 2:

```

../../node_modules/@types/workbox-build/_types.d.ts:1:10 - error TS2305: Module '"../../../../../../../Users/au87xu/mdrnwb/web/node_modules/workbox-routing"' has no exported member 'RouteHandlerCallback'.

1 import { RouteHandlerCallback, RouteMatchCallback } from 'workbox-routing';
           ~~~~~~~~~~~~~~~~~~~~

../../node_modules/@types/workbox-build/_types.d.ts:1:32 - error TS2305: Module '"../../../../../../../Users/au87xu/mdrnwb/web/node_modules/workbox-routing"' has no exported member 'RouteMatchCallback'.

1 import { RouteHandlerCallback, RouteMatchCallback } from 'workbox-routing';
                                 ~~~~~~~~~~~~~~~~~~
```

Oh, typescript 🙃